python/pyspark/sql/dataframe.py

```diff
@@ -1830,6 +1830,43 @@ def agg(self, *exprs):
         """
         return self.groupBy().agg(*exprs)
 
+    @since(3.3)
+    def observe(self, observation, *exprs):
+        """Observe (named) metrics through an :class:`Observation` instance.
+
+        This method does not support streaming datasets.
+
+        .. versionadded:: 3.3.0
```

Comment:
Let's move this under "A user can retrieve ..."

Reply:
moved